src/App.tsx

```diff
@@ -1,24 +1,317 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { Todos } from './components/Todos';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
 
-const USER_ID = 0;
+import {
+  getTodos, postTodos, deleteTodo, patchTodos,
+} from './api/todos';
+
+import { Todo } from './types/Todo';
+import { FilteredBy } from './types/FilteredBy';
+import { Errors } from './types/Errors';
+
+const USER_ID = 9964;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [filteredBy, setFilteredBy] = useState<FilteredBy>(FilteredBy.All);
+  const [typeOfError, setTypeOfError] = useState<Errors>(Errors.None);
+  const [isTodoAdded, setTodoAdded] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [updatingTitle, setUpdatingTitle] = useState('');
+  const [targetTodosIds, setTargetTodosIds] = useState<number[]>([]);
+  const isActive = todos?.some(todo => !todo.completed);
+
+  async function getTodoList() {
+    try {
+      const todoList = await getTodos(USER_ID);
+
+      setTodos(todoList);
+    } catch (error) {
+      setTypeOfError(Errors.Loading);
+    }
+  }
+
+  const postNewTodo = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newTodo = {
+      id: 0,
+      title: newTodoTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    if (newTodoTitle !== '') {
+      setTodoAdded(true);
+      try {
+        const todo = await postTodos(USER_ID, newTodo);
+
+        setTodos((currTodo) => {
+          if (currTodo) {
+            return [...currTodo, ...todo];
```

Comment:
```suggestion
            return [...currTodo, todo];
```
after adding new todo app is failed
looks like problem here
![Screenshot from 2023-05-04 13-15-01](https://user-images.githubusercontent.com/33029864/236177055-590dab57-a234-4b22-83cb-ab2f62585e97.png)